examples/nitro.c

```diff
@@ -132,5 +204,12 @@ int main(int argc, char *const argv[])
         return -1;
     }
 
+    ret = pthread_join(thread, NULL);
+    if (ret < 0) {
+        perror("unable to join listener thread");
+        close(sock_fd);
+        exit(1);
+    }
+
```

Comment:
Is there ever a scenario where this gets reached? Shouldn't `krun_start_enter` either exit above or not at all?

Reply:
Usually yes, but nitro enclaves is a special case in this scenario (`krun_start_enter` does return). Not sure if that will remain true, but for now it is.

Reply:
Right, thanks for the clarification!